### ISSUE TITLE
Dialogs flicker and flash a dark shade right before they close in Next.js 15.

### ISSUE THREAD

**Issue by kamyarkaviani** [radix-ui/themes#636]
Demo:

https://github.com/user-attachments/assets/677ed304-83ae-4d70-b310-102db7660d57

**Steps to reproduce:**

1. Create a blank Next.js project with `npx create-next-app`
2. Follow instructions to install latest Next.js with TS/Eslint/No tailwind/No src/App router/No turbo
3. `npm i @radix-ui/themes`
4. Add the import `import '@radix-ui/themes/styles.css';` and the `Theme` provider in the root layout.
5. Change page.tsx to 
```
import { Button, Dialog } from "@radix-ui/themes";

export default function Home() {
  return (
    <div>
      <Dialog.Root>
        <Dialog.Trigger>
          <Button>test</Button>
        </Dialog.Trigger>
        <Dialog.Content>
          <Dialog.Title>title</Dialog.Title>
          <Dialog.Description>
            test
          </Dialog.Description>
        </Dialog.Content>
      </Dialog.Root>
    </div>
  );
}

```

The behavior persists with React 18 (`npm i react-dom@18 react@18`) and with different browsers. Same if you enable tailwind.

**Comment by kamyarkaviani** [comment]
This regression started at version `3.1.5`. `npm i @radix-ui/themes@3.1.4` fixes this flicker.

**Comment by chaance** [comment]
Fixed in #649
